manager/controllers/app/m4dapplication_controller.go

```diff
@@ -272,3 +299,26 @@ func (r *M4DApplicationReconciler) GetAllModules() (map[string]*app.M4DModule, e
 	}
 	return moduleMap, nil
 }
+
+// CreateAppNamespace creates a namespace in which the blueprint and the relevant resources will be running
+// It stores the generated namespace name inside app status
+func (r *M4DApplicationReconciler) CreateAppNamespace(app *app.M4DApplication) error {
+	genNamespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "m4d-"}}
+	genNamespace.Labels = make(map[string]string)
+	genNamespace.Labels["ownerName"] = app.Name
+	genNamespace.Labels["ownerNamespace"] = app.Namespace
+	if err := r.Create(context.Background(), genNamespace); err != nil {
+		return err
+	}
+	r.Log.V(0).Info("Created namespace " + genNamespace.Name + " for " + app.Namespace + "/" + app.Name)
+	app.Status.BlueprintNamespace = genNamespace.Name
+	return nil
+}
+
+// DeleteAppNamespace deletes the blueprint namespace upon blueprint deletion
+func (r *M4DApplicationReconciler) DeleteAppNamespace(name string) error {
```

Comment:
This is a generic `DeleteNamespace`, not tried to App. Do we have somewhere to move this to?

Reply:
Changed the name. I would leave this function here because it is going to be changed, together with the create function, when we will manage subNamespaceAnchors instead of namespaces (HNC related)